File: msgraph-mail/java/utilities/src/main/java/graphjavav4/utilities/users/mailFolders/messages/extensions/ExtensionsResponse.java
package graphjavav4.utilities.users.mailFolders.messages.extensions;

import com.microsoft.kiota.serialization.Parsable;
import com.microsoft.kiota.serialization.ParseNode;
import com.microsoft.kiota.serialization.SerializationWriter;
import graphjavav4.utilities.users.Extension;
import java.util.function.BiConsumer;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Objects;
public class ExtensionsResponse implements Parsable {
    @javax.annotation.Nullable
    public List<Extension> value;
    @javax.annotation.Nullable
    public String nextLink;
    public void serialize(@javax.annotation.Nonnull final SerializationWriter writer) {
        Objects.requireNonNull(writer);
        writer.writeCollectionOfObjectValues("value", value);
        writer.writeStringValue("nextLink", nextLink);
    }
    @javax.annotation.Nonnull
    public <T> Map<String, BiConsumer<T, ParseNode>> getDeserializeFields() {
        final Map<String, BiConsumer<T, ParseNode>> fields = new HashMap<>(2);
        fields.put("value", (o, n) -> { ((ExtensionsResponse)o).value = n.getCollectionOfObjectValues(Extension.class); });
        fields.put("nextLink", (o, n) -> { ((ExtensionsResponse)o).nextLink = n.getStringValue(); });
        return fields;
    }
}

File: msgraph-mail/go/utilities/users/mailfolders/messages/extensions/extensions_response.go
package extensions

import (
    i04eb5309aeaafadd28374d79c8471df9b267510b4dc2e3144c378c50f6fd7b55 "github.com/microsoft/kiota/abstractions/go/serialization"
    ieea96ea0706c7e10d110f01563f903230c17531f1ba4f5e7095035777bc8b5e5 "github.com/microsoft/kiota-samples/msgraph-mail/go/utilities/models"
)

type ExtensionsResponse struct {
    additionalData map[string]interface{};
    nextLink *string;
    value []ieea96ea0706c7e10d110f01563f903230c17531f1ba4f5e7095035777bc8b5e5.Extension;
}
func (m *ExtensionsResponse) getAdditionalData()(map[string]interface{}) {
    return nil
}
func (m *ExtensionsResponse) getNextLink()(*string) {
    return nil
}
func (m *ExtensionsResponse) getValue()([]ieea96ea0706c7e10d110f01563f903230c17531f1ba4f5e7095035777bc8b5e5.Extension) {
    return nil
}
func (m *ExtensionsResponse) getFieldDeserializers()(map[string]func(interface{}, i04eb5309aeaafadd28374d79c8471df9b267510b4dc2e3144c378c50f6fd7b55.ParseNode)(error), error) {
    return nil
}
func (m *ExtensionsResponse) serialize(writer i04eb5309aeaafadd28374d79c8471df9b267510b4dc2e3144c378c50f6fd7b55.SerializationWriter)(error) {
    return nil
}
func (m *ExtensionsResponse) setAdditionalData(value map[string]interface{})() {
    return nil
}
func (m *ExtensionsResponse) setNextLink(value *string)() {
    return nil
}
func (m *ExtensionsResponse) setValue(value []ieea96ea0706c7e10d110f01563f903230c17531f1ba4f5e7095035777bc8b5e5.Extension)() {
    return nil
}
